clippy_lints/src/methods/mod.rs

```diff
@@ -330,6 +330,24 @@ declare_clippy_lint! {
     "using `Option.map_or(None, f)`, which is more succinctly expressed as `and_then(f)`"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for usage of `_.map_or(None, Some)`.
+    ///
+    /// **Why is this bad?** Readability, this can be written more concisely as
+    /// `_.ok()`.
+    ///
+    /// **Known problems:**
+    ///
+    /// **Example:**
+    /// ```rust
+    /// # let opt = Some(1);
+    /// # let r = opt.map_or(None, Some);
```

Comment:
lines starting with `#` won't be shown. Please also add a bad and a good example

```suggestion
    /// # let opt = Some(1);
    /// let r = opt.map_or(None, Some);
```

Reply:
Fixed in 3a29aedf8db3af19ee0f64dee6f00489812e6cb0